### PR TITLE
Better error handling for errors in the telemetry service

### PR DESCRIPTION
**Pull request by cicdw**
### Summary

When the telemetry service is unavailable, the errors can look really messy and repeat consistently.  Because this doesn't affect application behavior, this PR shuts down the telemetry service when failing to ping.

### Checklist

This pull request is:

- [ ] A documentation / typographical error fix
	- No tests or issue needed
- [x] A short code fix
	- Please reference the related issue by including "closes `<link to issue>`" in this Pull Request's summary section. 
        - If no issue exists, please create a bug report issue 
	- Please include tests. One-line fixes without tests will not be accepted unless it's related to the documentation only.
- [ ] A new feature implementation
	- Please reference the related issue by including "closes `<link to issue>`" in this Pull Request's summary section. 
        - If no issue exists, please create a feature enhancement issue 
	- Please include tests
    - Please make sure that your QA steps are both thorough and easy to reproduce by someobdy with limited knowledge of the feature that you are submitting

**Happy engineering!**

### CODE REVIEW

**Review by dylanbhughes — APPROVED** [review]

**Review by chrisguidry — APPROVED** [review]
That looks right to me.  We _could_ setup a timer for it to eventually try again later, but this is a good stopgap

**Review by jlowin — DISMISSED** [review]
I don't think this will work -- the `stop()` method blocks until the service actually exits, so I think calling it from inside the service will infinite loop.

**Comment by jlowin** [comment]
@cicdw in https://github.com/PrefectHQ/prefect/pull/6131 I added a kwarg it so you can safely call `await self.stop(block=True)` from inside the service

File: src/prefect/orion/services/telemetry.py [hunk 1]
```diff
@@ -101,17 +102,21 @@ async def run_once(self):
...
+            # this looks like an impactful server error
+            self.logger.error(
+                "Failed to send telemetry.", exc_info=PREFECT_DEBUG_MODE.value()
             )
-
-            try:
-                result.raise_for_status()
-            except Exception:
-                self.logger.exception("Failed to send telemetry.")
+            await self.stop()
```
Comment (madkinsz) [inline src/prefect/orion/services/telemetry.py:119]:
```suggestion
            await self.stop(block=False)
```

**Comment by madkinsz** [comment]
Can we add a test for this?

**Comment by madkinsz** [comment]
@cicdw I've finished this, @dylanbhughes please review again

**Review by dylanbhughes — APPROVED** [review]

**Pull request merged**